### DISCUSSION TITLE
Help me configure a Strapi deployment in vercel

### DISCUSSION THREAD

**Discussion by fiderosado** [vercel/community#3098] [Help] [answered]
### Summary

I have tried in a thousand ways to be able to deploy a Strapi project in vercel without success, can someone guide me in the process

### Example

https://minormarket-admin.vercel.app/

### Steps to Reproduce

Configure a Strapi project local
Upload to github
Add project to vercel, configure the Env variables
Configure the postgre database
Adding Custom comands 
deploy
-- nothing happend

**Comment by motionharvest** [comment]
Bump, I need help on this too.

**Reply by fiderosado** [reply]
It has not worked, instead I have deployed my app on Render.com, I recommend it

**Comment by amyegan** [comment]
Hey there! Strapi has a [Strapi Next.js Corporate Website Starter](https://vercel.com/templates/next.js/strapi-nextjs-corporate-starter) template (and [repo](https://github.com/strapi/nextjs-corporate-starter)) that you can use as a model for project configuration. 

If any errors happen during deployment, you can check the [build logs](https://vercel.com/docs/deployments/logs) for more information. After a successful deployment, you can check the [runtime logs](https://vercel.com/docs/observability/runtime-logs) for additional information about serverless functions and page routes. If you share the error here, I can help you try to figure out what's causing it.

**Reply by fiderosado** [reply]
It has not worked, instead I have deployed my app on Render.com, I recommend it

**Reply by mpstaton** [reply]
@amyegan I already have worked super hard on my data model / content model locally.  I want to deploy Strapi to Vercel.  I keep getting dependency errors and thus have no production build.  I don't want a starter template.  That assumes I want to start from scratch.  That's an inaccurate assumption about many who would be in this circumstance.  

Just to give more context, Vercel is a sponsor of Strapi.  Everything I read say it should work seamlessly.  But Strapi isn't even a choice in the Framework stack.  But a bunch of other CMS' are.  If this is by design just say so.  Say something like "Vercel no longer recommends Strapi, use this other one."

**Reply by mpstaton** [reply]
@amyegan checking the "Build logs" and the "runtime logs" helps ZERO.

**Reply by amyegan** [reply]
Hey @mpstaton. You posted a lot of messages, but I'm going to try to respond to them all at once here to keep notification noise to a minimum for the other folks here.

I recommend using the starter project as a reference to compare its configuration with the project that has a deployment issue. The [template](https://vercel.com/templates/next.js/strapi-nextjs-corporate-starter) was created and is maintained by the Strapi team: https://github.com/strapi/nextjs-corporate-starter. The readme has a solid outline of how to deploy the frontend and backend for the app. You would definitely need to deploy the backend _before_ the frontend so you could have the necessary env var values available. 

Strapi also shared a guide to [deploy a Next.js blog starter using Strapi on Vercel](https://strapi.io/blog/strapi-online-meetup-4-recap) a few years ago. It works in a similar way, but in that example they deployed the Strapi backend on Heroku. They also have another blog post about the frontend and backend repos for their conference site that year: https://strapi.io/blog/virtual-event-starter-template-vercel-strapiconf.

The build error you shared looks like the kind of issue caused by deploying the root of repo that contains separate frontend and backend pieces, like in a [monorepo](https://vercel.com/docs/monorepos). But there's not enough info here for me to know exactly what is happening. If you share your repo or a [minimal reproducible example](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), someone here may be able recognize what's going wrong and propose a solution.

**Reply by mpstaton** [reply]
@amyegan thanks for such a thorough response. [Here is my public repo](https://github.com/mpstaton/contentGraph). Here is the [Vercel deployment](https://content-graph.vercel.app/). As for the minimal reproducible example, it's three lines in the command line -- npm install, npm run develop, npm run build.  So, please feel my gratitude before I overreact. 

This is the kind of explainer that should be painfully obvious in ALL documentation. The fact you are recommending a Strapi  / Next.js blog starter on Vercel two years ago shows me that neither Strapi nor Vercel care at all.  2 years ago means nothing works, period. 

As a matter of fact, there are Internet articles about how Vercel has a whole Strapi environment ready to go.  But when you are choosing frameworks, there are like 5 other CMS options but not Strapi.  Should I just abandon Strapi?  

I've been reading the documentation from start through, and NOWHERE did it show up to either deploy or not deploy Strapi as a monorepo, nowhere does it say to structure the directory like anything in particular.  NOWHERE are there instructions to deploy some file within the root file rather than the root file.  Given the way Vercel, Netlify, and Render work, which is to simply connect a repo, this sounds almost like someone is trying to sabotage people new to the ecosystem.  Thoughts?  

Vercel is marketing itself as kind of a simpler, more approachable version of AWS or Google Cloud or Azure.  Yet, here I am venting with no fix in sight.  

If you look at the start of the Docs of Strapi, and the start of the Docs for Vercel, they make it seem like if you just follow the instructions, magic will happen.  But I'm a couple days in trying to bank my head against the wall here.  I'm not saying this is your fault, but it's most definitely Vercel and Strapi's fault.  If two systems say they like each other, but they don't actually work, how am I supposed to interpret that?

**Comment by motionharvest** [comment]
I have abandoned Strapi for Contentful.

On Fri, Aug 25, 2023 at 4:02 PM Amy ***@***.***> wrote:

> Hey there! Strapi has a Strapi Next.js Corporate Website Starter
> <https://vercel.com/templates/next.js/strapi-nextjs-corporate-starter>
> template (and repo <https://github.com/strapi/nextjs-corporate-starter>)
> that you can use as a model for project configuration.
>
> If any errors happen during deployment, you can check the build logs
> <https://vercel.com/docs/deployments/logs> for more information. After a
> successful deployment, you can check the runtime logs
> <https://vercel.com/docs/observability/runtime-logs> for additional
> information about serverless functions and page routes. If you share the
> error here, I can help you try to figure out what's causing it.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/vercel/discussions/3098#discussioncomment-6826475>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFL2SP4L3QKHCUHGWROCE3XXEAGBANCNFSM6AAAAAAZ3T7BEU>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Aaron Sherrill*
Design Technologist
(203) 500- 9851 <2035009851>

**Reply by fiderosado** [reply]
It has not worked, instead I have deployed my app on Render.com, I recommend it

**Comment by mpstaton** [comment]
<img width="1680" alt="20240203__strapi-vercel--error" src="https://github.com/vercel/community/assets/5722/e18dce02-52e1-4ae1-82d8-89be96cf9373">

**Comment by mpstaton** [comment]
Well @amyegan , I went to deploy your NextJS starter kit and it assumes that I have a live version of Strapi already deployed.  Doesn't that kind of defeat the purpose?  I don't, there are no environment variables to enter. 
<img width="1422" alt="20240203__next-strapi-vercel--error" src="https://github.com/vercel/community/assets/5722/20cda4de-d234-44da-965c-6df3bee46e83">

**Comment by motionharvest** [comment, marked as answer]
What I found (Last August) was that Strapi can't actually be deployed and
hosted on Strapi. There isn't a backend node process that stays running
like on Heroku. So without writing a bunch of middleware for edge
functions, getting the backend and Frontend working on the same deployment
isn't possible.

I'd love to be proven wrong.

https://stackoverflow.com/questions/65935135/deploying-strapi-in-vercel

https://vercel.com/guides/using-express-with-vercel

https://vercel.com/docs/functions/edge-functions

Bummer, cause everywhere it says Here's a template where they are both
together. But what I think they mean is that you run Strapi locally and on
deploy you start up the server, build your app, then the server is turned
off and your app is static.

To make updates to your models you have to do it locally, which defeats the
purpose of a CMS.

Good luck, Aaron

**Reply by mpstaton** [reply]
This is so heartbreaking.  I've spent a very long time building out my data schema and entering data in my local Strapi.  I can't deploy it anywhere. Not even TO STRAPI CLOUD..

**Comment by motionharvest** [comment]
I was able to find this writeup on a serverless install of strapi (a little
old) https://blog.vikfand.com/posts/serverless-strapi-tutorial/ Same author
has this newer one about Strapi 4 on AWS Lambd
https://blog.vikfand.com/posts/serverless-strapi-4-guide/

Here in the Strapi Guide it says (
https://docs.strapi.io/dev-docs/faq#can-strapi-be-run-in-serverless-environments
)
[image: image.png]


I have no idea why there is a Strapi website starter template on Vercel
with all the indication that you can start both on deploy. Its very
frustrating.
https://vercel.com/templates/next.js/strapi-nextjs-corporate-starter should
either be removed, or a huge banner at the top that says that you can't
host Strapi on Vercel.

I wound up using Contentful.

On Wed, Feb 7, 2024 at 12:06 AM Michael Staton ***@***.***>
wrote:

> This is so heartbreaking. I've spent a very long time building out my data
> schema and entering data in my local Strapi. I can't deploy it anywhere.
> Not even TO STRAPI CLOUD..
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/vercel/discussions/3098#discussioncomment-8390935>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAFL2SM5WQ3BWWZS6MZ4OFDYSMDTXAVCNFSM6AAAAAAZ3T7BEWVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4DGOJQHEZTK>
> .
> You are receiving this because you commented.Message ID:
> ***@***.***>
>


-- 
*Aaron Sherrill*
Design Technologist
(203) 500- 9851 <2035009851>

**Reply by motionharvest** [reply]
If you have all your stuff set up and you want to use Strapi, deploy it to Heroku. https://docs.strapi.io/dev-docs/deployment/heroku
